src/Console/Commands/CashierUpdate.php

```diff
@@ -36,7 +42,38 @@ public function handle()
 
         $this->comment('Publishing Cashier v2 migrations...');
         $this->callSilent('vendor:publish', ['--tag' => 'cashier-update']);
+        sleep(2);
+        //Enable lock on Cashier::run()
 
+
+        $this->comment('Lock Cashier::run()');
+        $this->lock(CashierRun::class);
+
+        //Clone impacted tables with cashier_backup prefix
+        $this->comment('Clone impacted tables with cashier_backup prefix');
+        DB::statement('CREATE TABLE cashier_backup_orders LIKE orders');
+        DB::statement('INSERT cashier_backup_orders SELECT * FROM orders');
+
+        //Apply new structure to tables
+        $this->comment('Migrate database');
+        $this->callSilent('migrate');
+
+        //Import Mollie payments, first payment actions and refunds.
+        $paymentsIds = Order::all()->pluck('mollie_payment_id');
+        $mollieApi = Mollie::api();
+
+        $this->comment('Import Mollie payments, first payment actions');
+        $paymentsIds->each(function ($payment) use ($mollieApi) {
+            $molliePaymentData = $mollieApi->payments()->get($payment);
+            $ownerType = $molliePaymentData->metadata->owner->type;
+            $ownerID = $molliePaymentData->metadata->owner->id;
+            $owner = $ownerType::findOrFail($ownerID);
+            Payment::createFromMolliePayment($molliePaymentData, $owner);
+        });
```

Comment:
May need to wrap this into a DB transaction. CashierRun is blocked, but the Mollie webhook can still get called, modifying the data.

Reply:
I add --maintenance option for update command